src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs

```diff
@@ -77,7 +99,7 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
                         string[] values = strValue.Split(s_separators);
                         foreach (string v in values)
                         {
```

Comment:
All `v` will have the same EnumType. And all members of the EnumType will have the same underlying type. So would it be better to do something like


We have EnumType, all members of the enum must have the same underlying type, is it more straightforward to check the enum type itself with `Enum.GetUnderlyingType(EnumType) == typeof(UInt64)` ? 

```c#
                     bool isUint64 = Enum.GetUnderlyingType(EnumType) == typeof(UInt64);
                    if (strValue.IndexOf(',') != -1)
                     {
                         long convertedValue = 0;
                         string[] values = strValue.Split(s_separators);
                         foreach (string v in values)
                         {
                                         convertedValue |= isUint64 ? unchecked((long)Convert.ToUInt64(v, culture) : Convert.ToInt64(v, culture);
                        }
```

Reply:
I am not sure how perf sensitive `EnumConverter.ConvertFrom()` is.

Reply:
Let's add field `bool myIsUInt64` and initialize the field `myIsUInt64 = Enum.GetUnderlyingType(EnumType) == typeof(UInt64)` in constructor. It's the best performance decision.

Reply:
Rewrite the PR, but without new field.